src/190B20.c

```diff
@@ -1471,9 +1507,38 @@ INCLUDE_ASM(s32, "190B20", func_80266F60);
 
 INCLUDE_ASM(s32, "190B20", func_80266F8C);
 
-INCLUDE_ASM(s32, "190B20", func_80266FD8);
+void func_80266FD8(ActorPart* part, s32 arg1) {
+    DecorationTable* decorationTable;
+
+    if ((part->idleAnimations != NULL) && !(part->flags & 2)) {
+        decorationTable = part->decorationTable;
+        if (decorationTable->unk_764 != arg1) {
+            decorationTable->unk_764 = arg1;
+            decorationTable->unk_766 = 0;
+            decorationTable->unk_765 = 1;
+        }
+    }
+}
 
-INCLUDE_ASM(s32, "190B20", func_80267018);
+void func_80267018(Actor* actor, s32 arg1) {
+    ActorPart* actorPartsTable;
+    ActorPart* actorNextPart;
+    ActorPart* actorPartTemp;
+    s32 flags;
+
+    actorPartsTable = actor->partsTable;
+    if (actorPartsTable != NULL) {
+        do {
+            flags = actorPartsTable->flags;
+            if (!(flags & 0x100001) && (actorPartsTable->decorationTable != NULL) && !(flags & 2) && (actorPartsTable->idleAnimations != NULL)) {
+                func_80266FD8(actorPartsTable, arg1);
+            }
+            actorPartTemp = actorPartsTable->nextPart;
```

Comment:
This temps (actorNextPart and actorPartTemp) shouldn't be necessary. You should be able to do actorPartsTable = actorPartsTable->nextPart

that being said, I think you should change the name of this temp since it's not really used as the table but is just an iterator for the parts inside a table. so maybe just `actorPart`.

and on the first line where you initialize it, you technically want the first one in the list, so it'd be `actorPart = &actor->partsTable[0];`. You may or may not need the & there - I don't remembe the type of the table exactly